flutter/shell/platform/tizen/nui_autofill_popup.cc

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/tizen/nui_autofill_popup.h"
+
+#include <dali-toolkit/dali-toolkit.h>
+#include <dali-toolkit/devel-api/controls/table-view/table-view.h>
+#include <dali-toolkit/public-api/controls/text-controls/text-label.h>
+
+#include "flutter/shell/platform/tizen/tizen_autofill.h"
+
+namespace flutter {
+
+bool NuiAutofillPopup::Touched(Dali::Actor actor,
+                               const Dali::TouchEvent& event) {
+  const Dali::PointState::Type state = event.GetState(0);
+  if (Dali::PointState::DOWN == state) {
+    std::string text =
+        actor.GetProperty(Dali::Actor::Property::NAME).Get<std::string>();
+    on_commit_(text);
+    Hide();
+  }
+  return true;
+}
+
+void NuiAutofillPopup::Hide() {
+  // TODO(Swanseo0) : There is a phenomenon where white traces remain for a
+  // while when popup disappears.
+  popup_.SetDisplayState(Dali::Toolkit::Popup::HIDDEN);
+}
+
+void NuiAutofillPopup::Hidden() {
+  popup_.Unparent();
+  popup_.Reset();
+}
+
+void NuiAutofillPopup::OutsideTouched() {
+  Hide();
+}
+
+void NuiAutofillPopup::PrepareAutofill() {
+  popup_ = Dali::Toolkit::Popup::New();
+  popup_.SetProperty(Dali::Actor::Property::NAME, "popup");
+  popup_.SetProperty(Dali::Actor::Property::PARENT_ORIGIN,
+                     Dali::ParentOrigin::TOP_LEFT);
+  popup_.SetProperty(Dali::Actor::Property::ANCHOR_POINT,
+                     Dali::AnchorPoint::TOP_LEFT);
+  popup_.SetProperty(Dali::Toolkit::Popup::Property::TAIL_VISIBILITY, false);
+  popup_.SetBackgroundColor(Dali::Color::WHITE_SMOKE);
+  popup_.OutsideTouchedSignal().Connect(this,
+                                        &NuiAutofillPopup::OutsideTouched);
+  popup_.HiddenSignal().Connect(this, &NuiAutofillPopup::Hidden);
```

Comment:
Hmm. `Hide` and `Hidden` sound too similar. If I understand correctly, if `Hide` is called, the popup is closed (hid), and in turn the `Hidden` callback is called. Is this right?

One option is to remove the `Hide` method and inline the implementation within `Touched`.

Reply:
> Hmm. `Hide` and `Hidden` sound too similar. If I understand correctly, if `Hide` is called, the popup is closed (hid), and in turn the `Hidden` callback is called. Is this right?

Correct.

> One option is to remove the Hide method and inline the implementation within Touched.

I think it's a good option. I will modify it.